src/main/scala-2.12/org/apache/james/gatling/imap/scenari/ImapFullScenario.scala

```diff
@@ -33,24 +32,24 @@ class ImapFullScenario {
   val lightUser = receiveEmail
     .exec(readLastEmail)
 
-  def generate(duration: Duration, users: Seq[Future[User]]): ScenarioBuilder =
+  def generate(duration: Duration, feeder: FeederBuilder): ScenarioBuilder =
     scenario("Imap")
-      .feed(UserFeeder.createCompletedUserFeederWithInboxAndOutbox(users))
+      .feed(feeder)
       .pause(1.second)
       .during(duration) {
```

Comment:
put it later, in place of `forever()`s

Reply:
It need to be before the login command, otherwise the username and password session variable will not be found.

Reply:
What about this:

```
         exec(imap("Connect").connect()).exitHereIfFailed

          .exec(imap("login").login("${username}", "${password}").check(ok))
          .exec {
            randomSwitch(
              75.0 -> during(duration) {
                pace(5 minutes, 15 minutes)
                  .exec(lightUser)
              },

              25.0 -> during(duration) {
                pace(30 seconds, 90 seconds)
                  .exec(heavyUser)
              })
          }
```

Reply:
does it work to put the duration after the switch? We don't risk to define the probability before running?

Reply:
I think the last modification even if not perfect represent an improvement over the current situation.
To do what we really want we should decompose the ImapFullScenario into 2 scenario. One for the heavy users, one for the light ones.
And on the ImapFullSimulation run the light scenario with 75% of the users, the heavy one with 25% of the users.
This way the 2 scenario would run concurrently. 

Should i create a new issue for this ?

Reply:
ok, let's create an issue about this so that we can merge this ticket